src/app/routes/api/v3/forms/__tests__/public-forms.verification.routes.spec.ts

```diff
@@ -297,6 +386,113 @@ describe('public-forms.verification.routes', () => {
     })
 
+    it('should return 201 when using SingPass auth and user is logged in to SingPass', async () => {
+      // Arrange
+      const mockJwt = 'mockJwt'
+
+      jest.spyOn(SpOidcClient.prototype, 'verifyJwt').mockResolvedValueOnce({
+        userName: 'S1234567A',
+      })
+
+      // Act
+      const response = await request
+        .post(
+          `/forms/${mockSPFormId}/fieldverifications/${mockTransactionId}/fields/${mockEmailFieldId}/otp/generate`,
+        )
+        .send({
+          answer: 'mail@me.com',
+        })
+        // Note cookie is for SingPass, not CorpPass
+        .set('Cookie', [`jwtSp=${mockJwt}`])
+
+      // Assert
+      expect(response.status).toBe(StatusCodes.CREATED)
+      expect(response.text).toEqual(getReasonPhrase(StatusCodes.CREATED))
+    })
+
+    it('should return 201 when using CorpPass auth and user is logged in to CorpPass', async () => {
+      // Arrange
+      const mockJwt = 'mockJwt'
+
+      mockCpClient.verifyJWT.mockImplementationOnce((_jwt, cb) =>
+        cb(null, {
+          userName: 'S1234567A',
+          userInfo: 'MyCorpPassUEN',
+        }),
+      )
+
+      // Act
+      const response = await request
+        .post(
+          `/forms/${mockCPFormId}/fieldverifications/${mockTransactionId}/fields/${mockEmailFieldId}/otp/generate`,
+        )
+        .send({
+          answer: 'mail@me.com',
+        })
+        // Note cookie is for CorpPass, not SingPass
+        .set('Cookie', [`jwtCp=${mockJwt}`])
+
+      // Assert
+      expect(response.status).toBe(StatusCodes.CREATED)
+      expect(response.text).toEqual(getReasonPhrase(StatusCodes.CREATED))
+    })
+
+    it('should return 201 when using MyInfo auth and user is logged in to MyInfo', async () => {
+      // Arrange
+      await MyInfoHashModel.updateHashes(
+        MOCK_UINFIN,
+        mockMyInfoFormId,
+        {},
+        MOCK_COOKIE_AGE,
+      )
+      const cookie = JSON.stringify({
+        accessToken: 'mockAccessToken',
+        usedCount: 0,
+        state: MyInfoCookieState.Success,
+      })
+
+      // Act
+      const response = await request
+        .post(
+          `/forms/${mockMyInfoFormId}/fieldverifications/${mockTransactionId}/fields/${mockEmailFieldId}/otp/generate`,
+        )
+        .send({
+          answer: 'mail@me.com',
+        })
+        .set('Cookie', [
+          // The j: indicates that the cookie is in JSON
+          `${MYINFO_COOKIE_NAME}=j:${encodeURIComponent(cookie)}`,
+        ])
+
+      // Assert
+      expect(response.status).toBe(StatusCodes.CREATED)
+      expect(response.text).toEqual(getReasonPhrase(StatusCodes.CREATED))
+    })
+
+    it('should return 201 when using SGID auth and user is logged in to SGID', async () => {
+      // Arrange
+      const MOCK_JWT = 'jwt'
+      const MOCK_JWT_PAYLOAD = {
+        userName: 'S1234567A',
+      }
+
+      MockJwt.verify.mockReturnValue(MOCK_JWT_PAYLOAD)
```

Comment:
Unsure if this is the right way to integration test, thinking to instead generate a valid SGID JWT and pass that in the request.

Reply:
it's fine. we should mock all third party calls so they are not dependent on the state of third parties